nns/nns_contract.go

```diff
@@ -273,21 +287,16 @@ func Register(name string, owner interop.Hash160, email string, refresh, retry,
 	}
 
 	l := len(fragments)
-	tldKey := append([]byte{prefixRoot}, []byte(fragments[l-1])...)
+	tldKey := makeTLDKey(fragments[l-1])
 	ctx := storage.GetContext()
 	tldBytes := storage.Get(ctx, tldKey)
 	if l == 1 {
 		checkCommittee()
 		if tldBytes != nil {
 			panic("TLD already exists")
 		}
-		storage.Put(ctx, tldKey, 0)
-		putNameStateWithKey(ctx, getTokenKey([]byte(name)), NameState{
-			Name: name,
-			// NNS expiration is in milliseconds
-			Expiration: int64(runtime.GetTime() + expire*1000),
-		})
-		putSoaRecord(ctx, name, email, refresh, retry, expire, ttl)
+		putTLD(ctx, tldKey)
+		saveDomain(ctx, name, email, refresh, retry, expire, ttl, nil)
```

Comment:
Is it OK to have `nil` instead of an empty `interop.Hash160`?

Reply:
yes, `checkAdmin` checks `len(NameState.Owner) == 0` so IIUC there is no difference b/w `nil` and `{}`